--- a/python/private/python.bzl
+++ b/python/private/python.bzl
@@ -625,7 +625,7 @@
         available_versions[tag.python_version] = {}

     if tag.coverage_tool:
-        available_versions[tag.python_version].setdefault("coverage_tool", {})[tag.platform] = tag.coverage_tool
+        available_versions[tag.python_version].setdefault("coverage_tool", {})[tag.platform] = str(tag.coverage_tool)
     if tag.patch_strip:
         available_versions[tag.python_version].setdefault("patch_strip", {})[tag.platform] = tag.patch_strip
     if tag.patches:
