--- a/src/rp2_common/pico_platform_sections/include/pico/platform/sections.h
+++ b/src/rp2_common/pico_platform_sections/include/pico/platform/sections.h
@@ -154,6 +154,15 @@
  * \param group a string suffix to use in the section name to distinguish groups that can be linker
  *              garbage-collected independently
  */
+#ifndef PICO_COMBINED_DOCS
+#define PICO_COMBINED_DOCS 0
+#endif
+#ifndef PICO_XIP_RAM
+#define PICO_XIP_RAM 0
+#endif
+#ifndef PICO_USE_XIP_CACHE_AS_RAM
+#define PICO_USE_XIP_CACHE_AS_RAM 0
+#endif
 #ifndef __in_xip_ram
 #if PICO_USE_XIP_CACHE_AS_RAM || PICO_COMBINED_DOCS
 #define __in_xip_ram(group) __attribute__((section(".xip_ram." group)))
@@ -198,6 +207,9 @@
  *
  * \param name  the name of the variable to place in the section
  */
+#ifndef LIB_PICO_LOW_POWER
+#define LIB_PICO_LOW_POWER 0
+#endif
 #if (LIB_PICO_LOW_POWER && HAS_POWMAN_TIMER) || PICO_COMBINED_DOCS
 #ifndef __persistent_data
 #define __persistent_data(name) __attribute__((section(".persistent_data." #name))) name
